app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,414 @@
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import org.kiwix.kiwixmobile.local_file_transfer.WifiDirectManager.Companion.getDeviceStatus
+import org.kiwix.kiwixmobile.local_file_transfer.adapter.WifiP2pDelegate
+import org.kiwix.kiwixmobile.local_file_transfer.adapter.WifiPeerListAdapter
+import java.util.ArrayList
+import java.util.Collections
+import javax.inject.Inject
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using [PeerGroupHandshakeAsyncTask]
+ * 2) After handshake, starting the files transfer using [SenderDeviceAsyncTask] on the sender
+ * device and [ReceiverDeviceAsyncTask] files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning")
+class LocalFileTransferActivity : BaseActivity(),
+  WifiDirectManager.Callbacks {
+  @Inject
+  var alertDialogShower: AlertDialogShower? = null
+
+  @Inject
+  var wifiDirectManager: WifiDirectManager? = null
+
+  @Inject
+  var locationManager: LocationManager? = null
+
+  @BindView(R.id.toolbar)
+  var actionBar: Toolbar? = null
+
+  @BindView(R.id.text_view_device_name)
+  var deviceName: TextView? = null
+
+  @BindView(R.id.progress_bar_searching_peers)
+  var searchingPeersProgressBar: ProgressBar? = null
+
+  @BindView(R.id.list_peer_devices)
+  var peerDeviceList: RecyclerView? = null
+
+  @BindView(R.id.text_view_empty_peer_list)
+  var textViewPeerDevices: TextView? = null
+
+  @BindView(R.id.recycler_view_transfer_files)
+  var filesRecyclerView: RecyclerView? = null
+  private var isFileSender = false // Whether the device is the file sender or not
+  private var filesForTransfer = ArrayList<FileItem>()
+  private var fileListAdapter: FileListAdapter? = null
+  private var wifiPeerListAdapter: WifiPeerListAdapter? = null
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_local_file_transfer)
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    val filesIntent = intent
+    val fileUriArrayList: ArrayList<Uri>?
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM)
+    isFileSender = fileUriArrayList != null && fileUriArrayList.size > 0
```

Comment:
using more kotlin features this would be
`fileUriArrayList?.isNotEmpty()==true`